src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/ExposureDetectionViewModel.swift

```diff
@@ -263,7 +268,16 @@ class ExposureDetectionViewModel: CountdownTimerDelegate {
 		case .low:
 			dynamicTableViewModel = lowRiskModel(risk: risk)
 		case .high:
-			dynamicTableViewModel = highRiskModel(risk: risk)
+			appConfigurationProvider.appConfiguration()
+				.sink { [weak self] in
+					let survayParameters = $0.eventDrivenUserSurveyParameters.common
+					let isSurvayEnabled = survayParameters.surveyOnHighRiskEnabled && !survayParameters.surveyOnHighRiskURL.isEmpty
+					self?.surveyOnHighRiskURL = survayParameters.surveyOnHighRiskURL
+					if let dynamicViewModel = self?.highRiskModel(risk: risk, isSurvayEnabled: isSurvayEnabled) {
+						self?.dynamicTableViewModel = dynamicViewModel
+					}
```

Comment:
```suggestion
				.sink { [weak self] in
				guard let self = self else {
				    Log.debug("failed to get strong self")
				    return
					let survayParameters = $0.eventDrivenUserSurveyParameters.common
					let isSurvayEnabled = survayParameters.surveyOnHighRiskEnabled && !survayParameters.surveyOnHighRiskURL.isEmpty
					self.surveyOnHighRiskURL = survayParameters.surveyOnHighRiskURL
					self.dynamicTableViewModel = self.highRiskModel(risk: risk, isSurvayEnabled: isSurvayEnabled) 
```
better like this? Please try out before submit change.